exo-run/src/service-restarter.ls

```diff
@@ -30,19 +31,21 @@ class ServiceRestarter extends EventEmitter
 
   _restart: ->
     @watcher.close!
-    cwd = @docker-config-location
-    DockerCompose.kill-container {service-name: @role, cwd, @write}, (exit-code) ~>
-      | exit-code => return @emit 'error', "Docker failed to kill container #{@role}"
-      @write "Docker container stopped"
-
-      DockerCompose.create-new-container {service-name: @role, cwd, @env, @write}, (exit-code) ~>
-        | exit-code => return @emit 'error', "Docker image failed to rebuild #{@role}"
-        @write "Docker image rebuilt"
-
-        DockerCompose.start-container {service-name: @role, cwd, @env, @write}, (exit-code) ~>
-          | exit-code => return @emit 'error', "Docker container failed to restart #{@role}"
-          @watch!
-          @logger.log {role: 'exo-run', text: "'#{@role}' restarted successfully"}
+    set-timeout ( ~>
```

Comment:
I have separated the delay variables and made the watcher use the debounced version of `_restart` but I'm still waiting for everyone to agree on actual delay times. Now that we're using debounced restart, I don't know if there is a need for this `set-timeout` function anymore. The point was to kill the watcher for the service directory and make the restarter wait for `@time-delay` and restart only once with the changes made (to all other files) during those couple seconds of the initial file change. Actually, I'm not entirely sure how this is different from debounced restart in terms of the final effect if the wait times are equal.